### PR TITLE
Match aux_loss dtype to ce_loss in SemanticSegmentationLoss

### PR DESCRIPTION
**Pull request by ahmet-f-gumustas**
## Summary

`SemanticSegmentationLoss.forward` builds the auxiliary-loss placeholder with `torch.tensor(0.0, device=preds.device)`, which always allocates a float32 scalar. When the surrounding losses are produced under a different dtype path (e.g. AMP), the three components fed into `torch.stack([ce_loss, dice_loss, aux_loss])` no longer share a dtype. Today PyTorch silently promotes, but this is fragile — any future tightening of stack's dtype rules, or a change in `_ce_loss`/`_dice_loss` casting, would surface here as a runtime error.

This PR pins the placeholder to `ce_loss.dtype` so `loss_items` has a single, predictable dtype.

## Change

```diff
- # Auxiliary cross-entropy loss.
- aux_loss = torch.tensor(0.0, device=preds.device)
+ # Auxiliary cross-entropy loss. Match ce_loss dtype so torch.stack below succeeds under AMP.
+ aux_loss = torch.tensor(0.0, device=preds.device, dtype=ce_loss.dtype)
```

Only [`ultralytics/utils/loss.py`](ultralytics/utils/loss.py) is touched.

## Test plan

- [x] Local sanity script exercising both `aux_logits is None` and the `(main, aux)` tuple branch — `loss_items` returned as a single-dtype tensor in both cases.
- [ ] Existing semantic CI (`tests/test_engine.py::test_task[semantic-...]`) continues to pass.

I have read the CLA Document and I sign the CLA

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Fixes a mixed-precision edge case in `SemanticSegmentationLoss` by aligning `aux_loss` dtype with `ce_loss` so loss aggregation works reliably under AMP. ⚙️

### 📊 Key Changes
- Updates the auxiliary loss initialization in `ultralytics/utils/loss.py` to use `dtype=ce_loss.dtype`.
- Adds an inline comment clarifying that this prevents `torch.stack` failures during AMP execution.
- Leaves the existing auxiliary loss computation logic unchanged when `aux_logits` is present.

### 🎯 Purpose & Impact
- Prevents dtype mismatches between loss tensors in semantic segmentation training. ✅
- Improves stability for AMP and mixed-precision workflows. 🚀
- Delivers a minimal, low-risk bug fix with no expected behavior change outside the affected edge case.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ahmet-f-gumustas, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness, and an Ultralytics engineer will assist you shortly 😊 Please review the checklist below to help ensure a smooth integration:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean, targeted fix. Matching `aux_loss` to `ce_loss.dtype` addresses the mixed-precision edge case without changing the auxiliary-loss path when `aux_logits` is present, and I don't see any regressions in the shown diff.

**Comment by Laughing-q** [comment]
@ahmet-f-gumustas nice spot! merging! thank you for your contribution
